frontend/templates/product/sections/ProductSection/ProductOptions.tsx

```diff
@@ -0,0 +1,223 @@
+                     <SimpleGrid columns={2} spacing="2">
+                        {option.values.map((value) => {
+                           const variant = findVariant(product.variants, {
+                              ...selectedOptions,
+                              [option.name]: value,
+                           });
+                           return (
+                              <ProductOptionValue
+                                 key={value}
+                                 isActive={variant?.id === selected}
+                                 label={value}
+                                 image={variant?.image}
+                                 onClick={() => {
+                                    if (variant) {
+                                       onChange(variant.id);
+                                    }
+                                 }}
+                              />
+                           );
+                        })}
+                     </SimpleGrid>
+                  )}
+               </VStack>
+            );
+         })}
+      </VStack>
+   );
+}
+
+enum SelectorType {
+   SINGLE,
+   SELECT,
+   RADIO,
+   IMAGE_RADIO,
+}
+
+type Variant = Product['variants'][0];
+
+type Option = Product['options'][0];
+
+function findVariant(
+   variants: Variant[],
+   selectedOptions: Record<string, string>
+) {
+   return variants.find((variant) => {
+      return variant.selectedOptions.every((option) => {
+         return selectedOptions[option.name] === option.value;
+      });
+   });
+}
+
+function getSelectorType(option: Option): SelectorType {
+   if (option.values.length === 1) {
+      return SelectorType.SINGLE;
+   }
+   if ([''].includes(option.name)) {
+      return SelectorType.RADIO;
+   }
+   if (['Part or Kit'].includes(option.name)) {
+      return SelectorType.IMAGE_RADIO;
+   }
+   return SelectorType.SELECT;
+}
+
+type ProductOptionProps = {
+   label: string;
+   image?: Image | null;
+   isActive?: boolean;
+   onClick?: () => void;
+};
+
+type Image = {
+   id?: string | null;
+   url: string;
+   altText?: string | null;
+   width?: number | null;
+   height?: number | null;
+};
+
+function ProductOptionValue({
+   label,
+   image,
+   isActive,
+   onClick,
+}: ProductOptionProps) {
+   return (
+      <Box
+         bg="white"
+         borderWidth={isActive ? 2 : 1}
+         borderColor={isActive ? 'brand.500' : 'gray.200'}
+         borderRadius="md"
+         px={image == null ? 2 : 2.5}
+         py={image == null ? 1.5 : 2.5}
+         cursor="pointer"
+         textAlign="center"
+         onClick={onClick}
+      >
+         {image && <ProductOptionImage image={image} />}
+         <Text fontSize="13px" color="gray.800">
+            {label}
+         </Text>
+      </Box>
+   );
+}
+
+type ProductOptionImageProps = {
+   image: Image;
+};
+
+function ProductOptionImage({ image }: ProductOptionImageProps) {
+   const ratio =
```

Comment:
Is it intentional that we're not using the `ratio` here, besides the null check?

Reply:
Oh good question. It's probably a bad refactor 🤔 Anyway next week @federicobadini will run through these to make them responsive, so I'll let him address this as he sees fit.

un_deploy_block 🏎️